retrievalmarket/types.go

```diff
@@ -371,6 +377,33 @@ func (dp *DealProposal) Type() datatransfer.TypeIdentifier {
 	return "RetrievalDealProposal/1"
 }
 
+// BindnodeSchema returns the IPLD Schema for a serialized DealProposal
+func (dp *DealProposal) BindnodeSchema() string {
+	return strings.Join([]string{
+		`type DealProposal struct {
+			PayloadCID &Any
+			ID Int # DealID
+			Params Params
+		}`,
+		(*Params)(nil).BindnodeSchema(),
+	}, "\n")
+}
+
+func DealProposalFromNode(node datamodel.Node) (*DealProposal, error) {
+	if node == nil {
+		return nil, fmt.Errorf("empty voucher")
+	}
+	dpIface, err := shared.FromNode(node, &DealProposal{})
+	if err != nil {
+		return nil, xerrors.Errorf("invalid DealProposal: %w", err)
+	}
+	dp, ok := dpIface.(*DealProposal)
+	if !ok {
+		return nil, fmt.Errorf("unexpected error, did not convert DealProposal")
```

Comment:
yeah, there's a few places this happens and it doesn't seem appropriate to continue blindly - although a panic might be better since it really would be a program error, maybe that's the right thing to do?